Honour `--no-deprecation` flag for deprecated rules (#7943)
stylelint/stylelint: docs/developer-guide/rules.md

Deprecating rules doesn't happen very often. When you do, you must:

1. Add the appropriate metadata to mark the rule as deprecated like so: `rule.meta = { deprecated: true }`.
2. Set the `stylelintType` to `'deprecation'`.
3. Optionally set `stylelintReference` to a link that points to a specific version of the rule's document so that it always remains accessible.

For example:

```js
result.warn(

>   `"your-namespace/old-rule" has been deprecated and will be removed in 7.0. Use '"your-namespace/new-rule" instead.`,

Author: ```suggestion
  `"your-namespace/old-rule" has been deprecated and will be removed in 7.0. Use "your-namespace/new-rule" instead.`,
```

Deprecating rules doesn't happen very often. When you do, you must:

> 1. Add the appropriate metadata to mark the rule as deprecated like so: `rule.meta = { deprecated: true }`
> 2. Set the `stylelintType` to `deprecation`

Reviewer: [suggest] Minor:

```suggestion
1. Add the appropriate metadata to mark the rule as deprecated like so: `rule.meta = { deprecated: true }`.
2. Set the `stylelintType` to `'deprecation'`.
```

> 3. Optionaly set `stylelintReference` to a link that points to a specific version of the rule's README on your source-code-hosting facility of choice—e.g. GitHub—so that it always remains accessible.

Author: see https://github.com/stylelint/stylelint/pull/6561#pullrequestreview-1248758992

Reviewer: [suggest] How about simplifying the description a bit more and adding an example to help people understand it? E.g.

````suggestion
3. Optionally set `stylelintReference` to a link that points to a specific version of the rule's document so that it always remains accessible.

For example:

```js
result.warn(
  `"your-plugin/old-rule" has been deprecated and will be removed in 7.0. Use '"your-plugin/new-rule" instead.`,
  {
    stylelintType: 'deprecation',
    stylelintReference: 'https://github.com/your-org/your-stylelint-plugin/blob/v6.3.0/src/rules/old-rule/README.md'
  },
);
```
````

Author: ````suggestion
3. Optionally set `stylelintReference` to a link that points to a specific version of the rule's document so that it always remains accessible.

For example:

```js
result.warn(
  `"your-namespace/old-rule" has been deprecated and will be removed in 7.0. Use '"your-namespace/new-rule" instead.`,
  {
    stylelintType: 'deprecation',
    stylelintReference: 'https://github.com/your-org/your-stylelint-plugin/blob/v6.3.0/src/rules/old-rule/README.md'
  },
);
```
````
